contract/src/agoric-basics.contract.js

```diff
@@ -20,78 +20,139 @@
+  return AmountMath.add(multiple, sum);
+};
+
+/**
+ *
  * @param {import('@endo/patterns').CopyBag} bag
- * @returns {bigint[]}
+ * @param {Inventory} inventory
+ * @returns {Amount}
  */
-const bagCounts = bag => {
+export const bagPrice = (bag, inventory) => {
+  /** @type {[string, bigint][]} */
   const entries = getCopyBagEntries(bag);
-  return entries.map(([_k, ct]) => ct);
+  const values = Object.values(inventory);
+  values.length > 0 || Fail`inventory must not be empty`;
+  const brand = values[0].tradePrice.brand;
+  return entries.reduce(
+    (sum, entry) => addMultiples(sum, entry, inventory),
+    // TODO: a better way to create empty amount
+    AmountMath.makeEmpty(brand),
+  );
 };
 // #endregion
 
+/** @typedef {{[key: string]: {tradePrice: Amount, maxTickets: NatValue}}} Inventory */
+
 /**
  * In addition to the standard `issuers` and `brands` terms,
  * this contract is parameterized by terms for price and,
- * optionally, a maximum number of items sold for that price (default: 3).
+ * optionally, a maximum number of tickets sold for that price (default: 3).
  *
  * @typedef {{
- *   tradePrice: Amount;
- *   maxItems?: bigint;
+ *   inventory: Inventory
  * }} AgoricBasicsTerms
  */
 
-export const meta = {
-  customTermsShape: M.splitRecord(
-    { tradePrice: AmountShape },
-    { maxItems: M.bigint() },
-  ),
-};
-// compatibility with an earlier contract metadata API
-export const customTermsShape = meta.customTermsShape;
-
 /**
  * Start a contract that
- *   - creates a new non-fungible asset type for Items, and
- *   - handles offers to buy up to `maxItems` items at a time.
+ *   - creates a new non-fungible asset type for Tickets, and
+ *   - handles offers to buy up to `maxTickets` tickets at a time.
  *
  * @param {ZCF<AgoricBasicsTerms>} zcf
  */
 export const start = async zcf => {
-  const { tradePrice, maxItems = 3n } = zcf.getTerms();
+  const { inventory } = zcf.getTerms();
 
   /**
-   * a new ERTP mint for items, accessed thru the Zoe Contract Facet.
+   * a new ERTP mint for tickets, accessed thru the Zoe Contract Facet.
    * Note: `makeZCFMint` makes the associated brand and issuer available
    * in the contract's terms.
    *
    * AssetKind.COPY_BAG can express non-fungible (or rather: semi-fungible)
    * amounts such as: 3 potions and 1 map.
    */
-  const itemMint = await zcf.makeZCFMint('Item', AssetKind.COPY_BAG);
-  const { brand: itemBrand } = itemMint.getIssuerRecord();
+  const ticketMint = await zcf.makeZCFMint('Ticket', AssetKind.COPY_BAG);
+  const { brand: ticketBrand } = ticketMint.getIssuerRecord();
+
+  const inventoryBag = makeCopyBag(
+    Object.entries(inventory).map(([ticket, { maxTickets }], _i) => [
+      ticket,
+      maxTickets,
+    ]),
+  );
+  const toMint = {
+    Tickets: {
+      brand: ticketBrand,
+      value: inventoryBag,
+    },
+  };
+  const inventorySeat = ticketMint.mintGains(toMint);
```

Comment:
Converted to `inventorySeat` approach.

Do I need to do anything to make sure `inventorySeat` persist between contract restarts/upgrades?

Reply:
If you were making the contract upgradable, yes, this is one of the things you'd have to keep track of.
(see https://github.com/Agoric/documentation/pull/968 )

Are we making this contract durable at this point? I didn't review it that way.

We talked about doing it as an enhancement to the swap contract
https://github.com/Agoric/documentation/issues/941#issuecomment-1915608319

But I'm open to whatever.

Reply:
Definitely not making this contract upgradable right now. More so planning out what things we can use down the line to teach contract upgrade.